app/frames/captcha/[slug]/route.tsx

```diff
@@ -0,0 +1,62 @@
+import { Button } from "frames.js/next";
+import { generateCaptchaChallenge } from "../../../../lib/captcha";
+import { frames } from "../../frames";
+import prisma from "../../../../lib/prisma";
+
+const frameHandler = frames(async (ctx) => {
+  // Check if the message is valid
+  if (!ctx?.message?.isValid) {
+    throw new Error("Invalid message");
+  }
+
+  // Get the waitlist slug
+  const slug = ctx.url.pathname.split("/").pop();
+
+  // Get the waitlist from the database
+  const waitlist = await prisma.waitlist.findUnique({
+    where: {
+      slug,
+    },
+  });
+
+  if (!waitlist) {
+    // TODO: show error frame
+    throw new Error("Invalid waitlist");
+  }
+
+  // Get the search params containing the user fid, ref and refSquared
+  const ref = ctx.url.searchParams.get("ref");
+  const refSquared = ctx.url.searchParams.get("refSquared");
```

Comment:
This shouldn't be here, it's not on main yet, it's only on Deca's branch

Reply:
I know, we are currently working on it together. If refSquared isn't found it simply isn't forwarded ahead. (So it should work totally fine in main as well)

Reply:
Branches are separated, so no point in keep this refSquared changes here